### PR TITLE
Add support for routes

### PR DESCRIPTION
**Pull request by andreyk-code**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +94,22 @@ The subnets list contains maps, where each object represents a subnet. Each map
...
+### Route Inputs
+The routes list contains maps, where each object represents a route. For the next_hop_* inputs, only one is possible to be used in each route. Having two next_hop_* inputs will produce an error. Each map has the following inputs (please see examples folder for additional references):
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| name | The name of the route being created  | string | - | no |
+| description | The description of the route being created | string | - | no |
+| tags | The network tags assigned to this route. This is a list in string format. Eg. "tag-01,tag-02"| string | - | yes |
+| dest_range | The destination range of outgoing packets that this route applies to. Only IPv4 is supported | string | - | yes 
+| next_hop_gateway | Whether the next hop to this route will the default internet gateway. Use "true" to enable this as next hop | string | - | yes |
```
Comment (morgante) [inline README.md]: I'd suggest calling this `next_hope_internet` instead.

File: README.md [hunk 2]
```diff
@@ -76,6 +94,22 @@ The subnets list contains maps, where each object represents a subnet. Each map
...
 
+### Route Inputs
+The routes list contains maps, where each object represents a route. For the next_hop_* inputs, only one is possible to be used in each route. Having two next_hop_* inputs will produce an error. Each map has the following inputs (please see examples folder for additional references):
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| name | The name of the route being created  | string | - | no |
+| description | The description of the route being created | string | - | no |
+| tags | The network tags assigned to this route. This is a list in string format. Eg. "tag-01,tag-02"| string | - | yes |
+| dest_range | The destination range of outgoing packets that this route applies to. Only IPv4 is supported | string | - | yes
```
Comment (morgante) [inline README.md]: Generally prefer the full `destination_range` over abbreviated names.

File: README.md [hunk 3]
```diff
@@ -42,6 +42,23 @@ module "vpc" {
 
         subnet-02 = []
     }
+    
+    routes = [
+        {
+            name             = "egress-inet"
```
Comment (morgante) [inline README.md]: Let's avoid (potentially confusing) abbreviations and call this `egress-internet`.

File: examples/simple_project/main.tf [hunk 4]
```diff
@@ -44,4 +44,21 @@ module "test-vpc-module" {
 
     subnet-02 = []
   }
+
+  routes = [
```
Comment (morgante) [inline examples/simple_project/main.tf]: I think we can leave routes out of the simple example.

File: main.tf [hunk 5]
```diff
@@ -42,9 +50,23 @@ resource "google_compute_subnetwork" "subnetwork" {
...
+  tags                   = "${compact(split(",",lookup(var.routes[count.index], "tags","")))}"
+  dest_range             = "${lookup(var.routes[count.index], "dest_range","")}"
+  next_hop_gateway       = "${lookup(var.routes[count.index], "next_hop_gateway","") == "true" ? "default-internet-gateway":""}"
+  next_hop_ip            = "${lookup(var.routes[count.index], "next_hop_ip","")}"
+  next_hop_instance      = "${lookup(var.routes[count.index], "next_hop_instance","")}"
+  next_hop_instance_zone = "${lookup(var.routes[count.index], "next_hop_instance_zone","")}"
+  next_hop_vpn_tunnel    = "${lookup(var.routes[count.index], "next_hop_vpn_tunnel","")}"
+  priority               = "${lookup(var.routes[count.index], "priority", "1000")}"
+
+  depends_on = ["google_compute_network.network"]
```
Comment (morgante) [inline main.tf]: Make sure linting doesn't return any errors (and run `terraform fmt`).

File: README.md [hunk 6]
```diff
@@ -42,6 +42,23 @@ module "vpc" {
...
+            dest_range       = "0.0.0.0/0"
+            tags             = "egress-inet"
+            next_hop_gateway = "true"
+        },
+        {
+            name        = "app-proxy"
+            description = "route through proxy to reach app"
+            dest_range  = "10.50.10.0/24"
+            tags        = "app-proxy"
+            next_hop_ip = "10.10.20.10"
```
Comment (morgante) [inline README.md]: Is using IPs considered a best practice for routing within GCP or should customers generally use instance names? Let's be prescriptive here.

**Comment by andreyk-code** [comment]
Thanks for review! Updated syntax per comments.

**Pull request merged**